### PR TITLE
Add com.github.SubhadeepJasu.pebbles

### PR DESCRIPTION
**Pull request by danilown**
### Please confirm your submission meets all the criteria

- [ OK] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [ OK] My pull request follows the instructions at [App Submission][submission].
- [ OK] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [ OK] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ OK] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ OK] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ OK] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.github.SubhadeepJasu.pebbles.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  - /share/valadoc
+  - /share/valadoc-0.48
+modules:
+  - name: gls
+    buildsystem: simple
+    build-commands:
+      - ./configure
+          --prefix=/app
+      - make
+      - make install
```
Comment (bilelmoussaoui) [inline com.github.SubhadeepJasu.pebbles.yaml:41]:
```suggestion
```
this should work fine as well

File: com.github.SubhadeepJasu.pebbles.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+runtime-version: '20.08'
+base: io.elementary.BaseApp
+base-version: juno-20.08
+command: com.github.subhadeepjasu.pebbles
+separate-locales: false
+finish-args:
+  - --share=network
+  - --socket=x11
+  - --socket=fallback-x11
+  - --socket=session-bus
```
Comment (bilelmoussaoui) [inline com.github.SubhadeepJasu.pebbles.yaml:13]: That's a bit too much, what exactly the app needs to talk to in the session bus? you can run the app with --log-session-bus to figure out which bus's it needs access to

File: com.github.SubhadeepJasu.pebbles.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+sdk: org.freedesktop.Sdk
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+base: io.elementary.BaseApp
+base-version: juno-20.08
+command: com.github.subhadeepjasu.pebbles
+separate-locales: false
+finish-args:
+  - --share=network
+  - --socket=x11
```
Comment (bilelmoussaoui) [inline com.github.SubhadeepJasu.pebbles.yaml:11]: You already have fallback-x11, you probably meant `--socket=wayland` here

File: com.github.SubhadeepJasu.pebbles.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+app-id: com.github.SubhadeepJasu.pebbles
+sdk: org.freedesktop.Sdk
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+base: io.elementary.BaseApp
+base-version: juno-20.08
+command: com.github.subhadeepjasu.pebbles
+separate-locales: false
```
Comment (bilelmoussaoui) [inline com.github.SubhadeepJasu.pebbles.yaml:8]: Any reason for this? 
```suggestion
```

File: com.github.SubhadeepJasu.pebbles.desktop [hunk 5]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
```
Comment (bilelmoussaoui) [inline com.github.SubhadeepJasu.pebbles.desktop:1]: You should use the desktop/appdata files from your upstream repository, there's no need to duplicate it here.

File: com.github.SubhadeepJasu.pebbles.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (bilelmoussaoui) [inline com.github.SubhadeepJasu.pebbles.appdata.xml:1]: Same as for the desktop file

File: com.github.SubhadeepJasu.pebbles.yaml (hunk 1 above)
Comment (danilown) [reply com.github.SubhadeepJasu.pebbles.yaml:41]: Indeed. I tested it locally and it works.

File: com.github.SubhadeepJasu.pebbles.yaml (hunk 2 above)
Comment (danilown) [reply com.github.SubhadeepJasu.pebbles.yaml:13]: I've disabled the `session-bus` and run it with `--log-session-bus` however I can't interpret the output.

```
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.com.github.SubhadeepJasu.pebbles
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for com.github.SubhadeepJasu.pebbles
C5: -> org.freedesktop.DBus fake ListNames
B1: <- org.freedesktop.DBus return from C1
B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B3: <- org.freedesktop.DBus return from C2
*SKIPPED*
B4: <- org.freedesktop.DBus return from C3
*SKIPPED*
B5: <- org.freedesktop.DBus return from C4
*SKIPPED*
B6: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Flatpak
C10: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B7: <- org.freedesktop.DBus return from C6
*SKIPPED*
B8: <- org.freedesktop.DBus return from C7
*SKIPPED*
B9: <- org.freedesktop.DBus return from C8
*SKIPPED*
B10: <- org.freedesktop.DBus return from C9
*SKIPPED*
B11: <- org.freedesktop.DBus return from C10
*SKIPPED*
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.RequestName at /org/freedesktop/DBus
Filtering message due to arg0 com.github.subhadeepjasu.pebbles, policy: 0 (required 3)
*HIDDEN* (ping)
B12: <- (no sender) return from C11
*REWRITTEN*
Failed to register: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown
```

File: com.github.SubhadeepJasu.pebbles.yaml (hunk 3 above)
Comment (danilown) [reply com.github.SubhadeepJasu.pebbles.yaml:11]: Actually the `--socket=fallback-x11` is unnecessary as I think the app doesn't work on Wayland yet.

Comment (bilelmoussaoui) [reply com.github.SubhadeepJasu.pebbles.yaml:11]: Then you need at least `--share=ipc`

Comment (hfiguiere) [reply com.github.SubhadeepJasu.pebbles.yaml:11]: It uses Gtk3, which uses Wayland, and I don't see where it forces X11 in the code. So what's the catch?

File: com.github.SubhadeepJasu.pebbles.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  - /share/devhelp
+  - /share/gir-1.0
+  - /share/gtk-doc
+  - /share/man
+  - /share/vala-0.40
+  - /share/vala-0.48
+  - /share/valadoc
+  - /share/valadoc-0.48
+modules:
+  - name: gls
```
Comment (hfiguiere) [inline com.github.SubhadeepJasu.pebbles.yaml:35]: typo: I think you meant `gsl`

(it doesn't really change anything to the build to be honest)

File: com.github.SubhadeepJasu.pebbles.yaml (hunk 3 above)
Comment (danilown) [reply com.github.SubhadeepJasu.pebbles.yaml:11]: Okay, I have added it to the manifest file.

**Comment by hfiguiere** [comment]
bot, build  com.github.SubhadeepJasu.pebbles

File: com.github.SubhadeepJasu.pebbles.yaml (hunk 4 above)
Comment (danilown) [reply com.github.SubhadeepJasu.pebbles.yaml:8]: No, removing it.

**Comment by hfiguiere** [comment]
bot, build com.github.SubhadeepJasu.pebbles

File: com.github.SubhadeepJasu.pebbles.yaml (hunk 3 above)
Comment (hfiguiere) [reply com.github.SubhadeepJasu.pebbles.yaml:10]: That mean having 
```
--socket=wayland
--socket=fallback-x11
--share=ipc
```

**Comment by hfiguiere** [comment]
Build fail because the appstream file isn't installed (or not found).


The one in the upstream source repository however has a different name (this is case sensitive), so if it is installed it's not found.

I recommend changing the app-id in the manifest to match upstream.

**Comment by hfiguiere** [comment]
That imply renaming the manifest file too.

**Comment by bilelmoussaoui** [comment]
> Build fail because the appstream file isn't installed (or not found).
> 
> The one in the upstream source repository however has a different name (this is case sensitive), so if it is installed it's not found.
> 
> I recommend changing the app-id in the manifest to match upstream.

That's probably the reason why you had `--socket=session-bus` because you don't own the app-id you mentioned in the flatpak manifest.

File: com.github.SubhadeepJasu.pebbles.yaml (hunk 7 above)
Comment (danilown) [reply com.github.SubhadeepJasu.pebbles.yaml:34]: Good catch!

**Comment by hfiguiere** [comment]
Using the upstream appdata and desktop files, means removing them from this repository, and changing the app-id in the manifest (and its file name) to match the app-id upstream.

**Comment by danilown** [comment]
> Using the upstream appdata and desktop files, means removing them from this repository, and changing the app-id in the manifest (and its file name) to match the app-id upstream.

Okay, I can remove the appdata and desktop files from this repository. 

However, the app-id already matches with upstream (if I got the app-id concept right). Here is it: https://github.com/SubhadeepJasu/pebbles

**Comment by bilelmoussaoui** [comment]
the app-id upstream is `com.github.subhadeepjasu.pebbles` & you're using `com.github.SubhadeepJasu.pebbles`

**Comment by hfiguiere** [comment]
bot, build com.github.subhadeepjasu.pebbles

File: com.github.SubhadeepJasu.pebbles.yaml (hunk 2 above)
Comment (bilelmoussaoui) [reply com.github.SubhadeepJasu.pebbles.yaml]: This can be dropped now :)

File: com.github.subhadeepjasu.pebbles.yaml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+runtime-version: '20.08'
+base: io.elementary.BaseApp
+base-version: juno-20.08
+command: com.github.subhadeepjasu.pebbles
+finish-args:
+  - --share=network
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=session-bus
```
Comment (barthalion) [inline com.github.subhadeepjasu.pebbles.yaml:13]:
```suggestion
```

**Comment by barthalion** [comment]
Should we invite someone beside you to the Flathub repo?

**Comment by danilown** [comment]
> Should we invite someone beside you to the Flathub repo?

Yes please, the owner of the upstream repository: https://github.com/SubhadeepJasu

File: com.github.subhadeepjasu.pebbles.yaml (hunk 8 above)
Comment (danilown) [reply com.github.subhadeepjasu.pebbles.yaml:13]: It won't work without it, please see https://github.com/flathub/flathub/pull/2263#discussion_r614365070 .

Comment (barthalion) [reply com.github.subhadeepjasu.pebbles.yaml:13]: It's discussed there that it will work now as app ID matches D-Bus ID.

Comment (danilown) [reply com.github.subhadeepjasu.pebbles.yaml:13]: Ah okay, sorry for the misunderstanding.

**Comment by barthalion** [comment]
bot, build com.github.subhadeepjasu.pebbles

**Comment by barthalion** [comment]
@bilelmoussaoui Consider adding cleanup script to your Elementary base apps. https://github.com/flathub/io.qt.qtwebengine.BaseApp/blob/branch/5.15/io.qt.qtwebengine.BaseApp.json#L87

/merge @SubhadeepJasu

**Comment by SubhadeepJasu** [comment]
Hey, thanks for making it available on Flathub. I also got a WIP branch "Odin" with flatpak builds. I'm going through all this now, I'll see what I can incorporate from here to that branch for future releases as well.

**Pull request closed without merging**